index.js

```diff
@@ -20,11 +20,15 @@ try {
     });
     define(proto, 'dispatchEvent', function (event) {
       var secret = wm.get(this);
-      var listeners = secret[event.type];
-      if (listeners) {
+      if (secret[event.type]) {
+        var listeners = secret[event.type].slice(0);
         define(event, 'target', this);
         define(event, 'currentTarget', this);
-        listeners.slice(0).forEach(dispatch, event);
```

Comment:
I would rollback this change and use `listeners.slice(0).some(dispatch, event)` and change `dispatch` to `return this.event._stopImmediatePropagationFlag;` to stop when that happens. That lets engines deal with `try/catches` too out of the box. Thoughts?

Reply:
very nice, I didn't think of that one 👍

Reply:
Pushed